lms/djangoapps/experiments/flags.py

```diff
@@ -54,6 +55,44 @@ def _cache_bucket(self, key, value):
         request_cache.set(key, value)
         return value
 
+    def _is_enrollment_inside_date_bounds(self, experiment_values, user, course_key):
+        from student.models import CourseEnrollment
+
+        enrollment_start = experiment_values.get('enrollment_start')
+        enrollment_end = experiment_values.get('enrollment_end')
+        if not enrollment_start and not enrollment_end:
+            return True  # early exit just to avoid any further lookups
+
+        now = datetime.datetime.now(pytz.utc)
+        enrollment = CourseEnrollment.get_enrollment(user, course_key)
+
+        # If the user isn't enrolled, act like they would enroll right now (this keeps the pre-enroll and post-enroll
+        # experiences the same, if they decide to enroll right now)
+        enrollment_creation_date = enrollment.created if enrollment else now
```

Comment:
This bit is new too -- treating non-enrolled users as if they were enrolled right now.

I was thinking like... if you see Dates tab before you enroll and then you click enroll and it's gone because the enrollment_start date hadn't happened yet, that's weird.

Reply:
Yeah, I think that's reasonable.